packages/relay/src/lib/eth.ts

```diff
@@ -1304,8 +1304,10 @@ export class EthImpl implements Eth {
         nonceCount = await this.getAccountLatestEthereumNonce(address, requestIdPrefix);
       } else if (blockNumOrTag === EthImpl.blockEarliest) {
         nonceCount = await this.getAccountNonceForEarliestBlock(requestIdPrefix);
-      } else if (!isNaN(blockNum)) {
+      } else if (!isNaN(blockNum) && blockNumOrTag.length != 66 && blockNum > 0) {
         nonceCount = await this.getAccountNonceForHistoricBlock(address, blockNum, requestIdPrefix);
+      } else if (blockNumOrTag.length == 66 && blockNumOrTag.startsWith('0x')) {
```

Comment:
Nit: Is there a reason 66 is not a descriptive numeric string literal? HASH_LENGTH

Reply:
we could also make a constant out of `0x`

Reply:
I've made the change 👍 
@ebadiere